core-bundle/src/Resources/contao/library/Contao/Files.php

```diff
@@ -156,6 +156,11 @@ public function fopen($strFile, $strMode)
 	{
 		$this->validate($strFile);
 
+		if ($strMode[0] !== 'r' && \dirname($strFile) && \dirname($strFile) !== '.')
+		{
+			$this->mkdir(\dirname($strFile));
+		}
+
```

Comment:
```suggestion
		if ($strMode[0] != 'r' && ($strPath = \dirname($strFile)) && $strPath != '.')
		{
			$this->mkdir($strPath);
		}

```

This might not be a problem at all due to the stat cache, but since `dirname()` calls are potentially expensive, it seems better to store the result in a variable.

Reply:
They are free in case of disabled open_basedir but we should change it anyway 👍